### PR TITLE
[luci] Use vector type input/output_type in QuantizeWithMinMaxPass

### PR DESCRIPTION
**Pull request by jinevening**
This uses vectoer type input/output_type in QuantizeWithMinMaxPass.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/10493
Draft PR: https://github.com/Samsung/ONE/pull/10518

### CODE REVIEW

**Comment by jinevening** [comment]
@nnfw-bot test nncc-jammy-debug

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp [hunk 1]
```diff
@@ -372,11 +372,22 @@ struct InsertQuantizeOp final : public luci::CircleNodeMutableVisitor<void>
...
+  assert(g); // FIX_CALLER_UNLESS
   auto inputs = g->inputs();
-  for (auto node : loco::input_nodes(g))
+
+  assert(inputs);                                    // FIX_CALLER_UNLESS
+  assert(inputs->size() == _ctx->input_type.size()); // FIX_CALLER_UNLESS
+
+  // NOTE loco::input_nodes returns input nodes following the order of InputIndex
+  auto input_nodes = loco::input_nodes(g);
+  for (uint32_t i = 0; i < input_nodes.size(); i++)
```
Comment (jinevening) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:383]: Changed from `for (auto node : loco::input_nodes(g))` to access index `i`. `i` is used to get `_ctx->input_type[i]`.

**Review by mhs4670go — DISMISSED** [review]
LGTM

**Comment by jinevening** [comment]
@seanshpark Ping

File: compiler/luci/pass/src/CircleQuantizer.cpp [hunk 2]
```diff
@@ -490,9 +490,8 @@ void CircleQuantizer::quantize(loco::Graph *g) const
       ctx->input_model_dtype = str_to_dtype(input_model_dtype);
       ctx->output_model_dtype = str_to_dtype(output_model_dtype);
       ctx->granularity = str_to_granularity(granularity);
-      // TODO Change input/output_type to vector
-      ctx->input_type = input_types[0];
-      ctx->output_type = output_types[0];
+      ctx->input_type = input_types;
```
Comment (seanshpark) [inline compiler/luci/pass/src/CircleQuantizer.cpp:493]: can we change `ctx->input_type` to `ctx->input_types` ?
`input_type = input_types` looks odd.

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp [hunk 3]
```diff
@@ -372,11 +372,22 @@ struct InsertQuantizeOp final : public luci::CircleNodeMutableVisitor<void>
 
 void QuantizeWithMinMaxPass::set_input_type(loco::Graph *g) const
 {
+  assert(g); // FIX_CALLER_UNLESS
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:375]: strictly speaking, split to another PR?

Comment (jinevening) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp]: I removed the change. PTAL

**Review by mhs4670go — APPROVED** [review]

**Review by seanshpark — APPROVED** [review]
LGTM
